includes/admin/post-types/class-ph-admin-cpt-contact.php

```diff
@@ -296,6 +296,75 @@ public function remove_bulk_actions( $actions ) {
         return $actions;
     }
 
+	/**
+	 * Search by email and phone number
+	 * Phone numbers are stripped of any none numeric or comma charactors
+	 *
+	 * @param string $where
+	 * @return string
+	 */
+	public function contact_search( $where ) {
+
+		global $pagenow, $wpdb, $wp;
+		
+		if ( 'edit.php' != $pagenow || ! is_search() || ! isset( $wp->query_vars['s'] ) || 'contact' != $wp->query_vars['post_type'] ) {
+			return $where;
+		}
+
+		if ( trim($wp->query_vars['s']) == '' )
+		{
+			return $where;
+		}
+
+		$search_ids = array();
+		$terms      = explode( ',', $wp->query_vars['s'] );
+
+		foreach ( $terms as $term )
+		{
+			if ( is_numeric( $term ) )
+			{
+				$search_ids[] = $term;
+			}
+
+            $phone_number = preg_replace( "/[^0-9,]/", "", $term );
+
+			// Attempt to get an ID by searching for phone and email address
+			$query = $wpdb->prepare( 
+				"SELECT 
+					ID
+				FROM 
+					{$wpdb->posts} 
+				INNER JOIN {$wpdb->postmeta} AS mt1 ON {$wpdb->posts}.ID = mt1.post_id
+				WHERE 
+					(
+						(mt1.meta_key='_telephone_number_clean' AND mt1.meta_value LIKE NULLIF(%s,'%%%'))
```

Comment:
Like this :) Not seen this NULLIF thing before and good to know 👍

Reply:
Super useful function :)